views/ThingiHeader.qml

```diff
@@ -15,91 +15,152 @@ Rectangle
 
     width: parent.width
     Layout.fillWidth: true
-    height: UM.Theme.getSize("toolbox_header").height
+    height: UM.Theme.getSize("toolbox_header").height * 2
     color: "#f5f5f5"
 
-    RowLayout
+    GridLayout
```

Comment:
I think the header menu gets a bit too complicated by doing this. Maybe instead it could be changed into a dropdown that holds all the menu items?